pkg/checks/health/health.go

```diff
@@ -75,8 +76,9 @@ func (h *Health) Run(ctx context.Context, cResult chan checks.ResultDTO) error {
 	ctx, cancel := logger.NewContextWithLogger(ctx)
 	defer cancel()
 	log := logger.FromContext(ctx)
-	log.Info("Starting healthcheck", "interval", h.config.Interval.String())
 
+	log.Info("Starting healthcheck", "interval", h.config.Interval.String())
+	h.Running = true
```

Comment:
The new changes address the concerns from @puffitos in https://github.com/caas-team/sparrow/pull/116#pullrequestreview-1907143614. I've simplified the `/healthz` endpoint by checking if every registered check is running. Therefore I've added a `IsRunning` method to the Checks interface so the internal state is taken into account.

Now the only question is: **When is a check running?** With how I implemented it now, we wouldn't address #79 because there is still a time difference between the check's `(*checks.Check).Run()` call and the first execution of each check's interval. So to give you an example:

Imagine having the health check configured with an interval of 10mins. Then the pod would be marked as ready even though the check hasn't run yet and will not run for another ~9,9min, resulting again in the problem of the issue:

![image](https://github.com/caas-team/sparrow/assets/27763017/6d0743ef-a543-434b-aa1c-9eab50ac33db)

To avoid this behavior, we could set the `Running` field of each check true after the first execution of the check but I that would again strongly connect the checks to the sparrow, which we didn't want to do.

Reply:
Let's discuss this in the issue first.